File: archive/pulumi-py/Pr6153/lib/lambda/payment_processor.py
"""Lambda function for processing payment transactions."""
import json
import os
import boto3
from datetime import datetime

dynamodb = boto3.resource('dynamodb')
table_name = os.environ['DYNAMODB_TABLE']
table = dynamodb.Table(table_name)


def handler(event, context):
    """Process payment transaction requests."""
    try:
        # Parse request body
        body = json.loads(event.get('body', '{}'))

        transaction_id = body.get('transactionId')
        amount = body.get('amount')
        currency = body.get('currency', 'USD')

        if not transaction_id or not amount:
            return {
                'statusCode': 400,
                'body': json.dumps({'error': 'Missing required fields'}),
            }

        # Store transaction in DynamoDB
        timestamp = int(datetime.now().timestamp())
        table.put_item(
            Item={
                'transactionId': transaction_id,
                'timestamp': timestamp,
                'amount': str(amount),
                'currency': currency,
                'status': 'pending',
            }
        )

        return {
            'statusCode': 200,
            'body': json.dumps({
                'message': 'Transaction processed',
                'transactionId': transaction_id,
                'timestamp': timestamp,
            }),
        }

    except Exception as e:
        print(f"Error processing transaction: {str(e)}")
        return {
            'statusCode': 500,
            'body': json.dumps({'error': 'Internal server error'}),
        }

File: archive/pulumi-ts/Pr7211/lib/lambda/payment-processor.js

exports.handler = async (event) => {
    console.log('Payment processing event:', JSON.stringify(event, null, 2));

    // Extract payment details from event
    const body = event.body ? JSON.parse(event.body) : event;

    return {
        statusCode: 200,
        headers: {
            'Content-Type': 'application/json',
            'Access-Control-Allow-Origin': '*'
        },
        body: JSON.stringify({
            message: 'Payment processed successfully',
            paymentId: body.paymentId || 'generated-id',
            status: 'completed',
            region: process.env.AWS_REGION,
            timestamp: Date.now()
        })
    };
};
